From 209f1b1357469aae7edc5b7ea2ef9ce380d51033 Mon Sep 17 00:00:00 2001
From: Rubio Lu <Rubio-DW.Lu@mediatek.com>
Date: Thu, 24 Feb 2022 16:51:33 +0800
Subject: [PATCH] mt76: remapping nl80211 DFS regions

Need to remap nl80211 DFS regions to chip regions definition
while initialing wiphy hw/interface or radar detector types
cannot be enabled

Change-Id: Ifc939fa21f3a09921db78cdf10ffd06c48d18cff
---
 mt7915/mac.c | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/mt7915/mac.c b/mt7915/mac.c
index 06186c03..0b36d093 100644
--- a/mt7915/mac.c
+++ b/mt7915/mac.c
@@ -2366,10 +2366,23 @@ static void mt7915_dfs_stop_radar_detector(struct mt7915_phy *phy)
 
 static int mt7915_dfs_start_rdd(struct mt7915_dev *dev, int chain)
 {
-	int err;
+	int err, region;
+
+	switch (dev->mt76.region) {
+	case NL80211_DFS_ETSI:
+		region = 0;
+		break;
+	case NL80211_DFS_JP:
+		region = 2;
+		break;
+	case NL80211_DFS_FCC:
+	default:
+		region = 1;
+		break;
+	}
 
 	err = mt76_connac_mcu_rdd_cmd(&dev->mt76, RDD_START, chain,
-				      MT_RX_SEL0, 0);
+				      MT_RX_SEL0, region);
 	if (err < 0)
 		return err;
 
-- 
2.18.0

